apps/forms/control-value-accessor/src/app/rating-control/rating-control.component.ts

```diff
@@ -1,23 +1,53 @@
-import { Component, EventEmitter, Output } from '@angular/core';
+import { Component,EventEmitter, Output } from '@angular/core';
+import {
+  AbstractControl,
+  ControlValueAccessor,
+  NG_VALUE_ACCESSOR,
+  ValidationErrors,
+  Validator,
+} from '@angular/forms';
+
 
 @Component({
   standalone: true,
   selector: 'app-rating-control',
   templateUrl: 'rating-control.component.html',
   styleUrls: ['rating-control.component.scss'],
+  providers: [
+    {
+      provide: NG_VALUE_ACCESSOR,
+      useExisting: RatingControlComponent,
+      multi: true,
+    },
+  ],
```

Comment:
You are missing this code to make the validation work:

```ts
    {
      provide: NG_VALIDATORS,
      multi: true,
      useExisting: RatingControlComponent,
    },
    ```

Reply:
Hey thanks for the review ...I had tested without inserting the  above code and validation did work. I realised that the above code works with template forms when implementing validation. Reactive forms seems to be working fine without the code. You could test it out and advise further